src/api/rate_limiter.py

```diff
@@ -119,6 +171,9 @@ async def make_request(self, url, cookies, method):
         await self.queue.put(request)
         await event.wait()
 
+        if "error" in self.requests[key]["result"]:
+            raise RuntimeError("Request failed")
+        
```

Comment:
I see what you're doing here by pasisng an `error` to the request result but isn't there a proper way to pass exceptions to this object from asyncio tasks loop? I could dig into it!